assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -126,7 +126,8 @@ def _raw_mdc_uri_folder_to_mltable(
     return table
 
 
-def _convert_mltable_to_spark_df(table: MLTable, preprocessed_input_data: str, fs: AbstractFileSystem) -> DataFrame:
+def _convert_mltable_to_spark_df(table: MLTable, preprocessed_input_data: str,
+                                 fs: AbstractFileSystem = None, add_tags_func=None) -> DataFrame:
```

Comment:
Why do we need to pass in the function?

Reply:
follow the param injection patten to inject a mock add_tags_func for testing